### PR TITLE
WT-12522 Add an option in Workgen to limit the number of files present in the database

### PR DESCRIPTION
**Pull request by etienneptl**
- Added a new setting `max_num_files` to limit the number of files present in the Workgn database
- Tables are no longer dynamically created if the limit has been reached

### CODE REVIEW

File: bench/workgen/workgen.cpp [hunk 1]
```diff
@@ -523,9 +545,17 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
...
           "key_format=S,value_format=S,app_metadata=\"" + DYN_TABLE_APP_METADATA);
 
+        uint32_t num_files = get_dir_num_files(_wt_home);
         int creates = 0, retries = 0;
-        while (
-          !stopping && creates < _workload->options.create_count && retries < TABLE_MAX_RETRIES) {
+        // Make sure not to exceed the maximum number of files that can exist in the database.
+        int num_files_to_create = 0;
+        if (num_files < max_files) {
+            if (num_files + _workload->options.create_count > max_files)
```
Comment (jiechenbo) [inline bench/workgen/workgen.cpp:553]: We can combine both the if statments of L552 and L553

File: bench/workgen/workgen.cpp [hunk 2]
```diff
@@ -523,9 +545,17 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
...
+                num_files_to_create = max_files - num_files;
+            else
+                num_files_to_create = _workload->options.create_count;
```
Comment (jiechenbo) [inline bench/workgen/workgen.cpp:556]: We should assert that the `create_count` has to be lower than the max_files variable.

File: bench/workgen/workgen.cpp (hunk 1 above)
Comment (etienneptl) [reply bench/workgen/workgen.cpp]: Thank you for spotting this: 47512934baf242b6ce98243698276cd139610324.

File: bench/workgen/workgen.cpp (hunk 2 above)
Comment (etienneptl) [reply bench/workgen/workgen.cpp]: Added a check in 3889e778b01f266914e8ed89d16646e79b535e99.

**Review by jiechenbo — APPROVED** [review]
LGTM!

File: bench/workgen/workgen.cpp [hunk 3]
```diff
@@ -476,40 +489,50 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
...
     if (manage_db_size) {
-        db_size = get_dir_size_mb(_wt_home);
-        // Initially we start creating tables if the database size is less than the create target.
-        creating = db_size < _workload->options.create_target;
+        uint32_t db_size = get_dir_size_mb(_wt_home);
+        uint32_t num_files = get_dir_num_files(_wt_home);
+        /*
+         * Initially we start creating tables if:
+         *  - the database size is less than the create target and
+         *  - then number of files is below the limit.
```
Comment (ruby-oujo) [inline bench/workgen/workgen.cpp:504]: should this be "the" not "then"

Comment (ruby-oujo) [inline bench/workgen/workgen.cpp]:
```suggestion
         *  - the number of files is below the limit.
```

Comment (etienneptl) [reply bench/workgen/workgen.cpp]: Thank you! Fixed in 75c489a864a596812c886b543791ff2ed68ce89d.

**Review by ruby-oujo — APPROVED** [review]

**Pull request merged**